homeassistant/components/tplink/light.py

```diff
@@ -474,6 +476,26 @@ def _set_device_state(self, state):
 
         return self._get_device_state()
 
+    async def async_update(self):
+        """Update the TP-Link bulb's state."""
+        for update_attempt in range(MAX_ATTEMPTS):
+            self._is_ready = False
```

Comment:
Rather than this variable, you could have `attempt_update` return the status. That would make it easier to follow.

Reply:
Make sense, I swear I originally did that way too and I might have had another reason for added the extra var but I don't see any reason for it either. Added change to next commit.